AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreConfigurationTests.swift

```diff
@@ -0,0 +1,150 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//
+//  DataStoreConfigurationTests.swift
+//  AWSDataStoreCategoryPluginIntegrationTests
+//
+//  Created by Guo, Rui on 6/15/20.
+//  Copyright © 2020 Amazon Web Services. All rights reserved.
+//
+
+import XCTest
+
+import AmplifyPlugins
+import AWSMobileClient
+import AWSPluginsCore
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSDataStoreCategoryPlugin
+
+class DataStoreConfigurationTests: XCTestCase {
+
+    func testConfigWithSameSchema() throws {
+
+        let databaseCreationDone = expectation(description: "Old database deleted and new database recreated")
+
+        do {
+            let dataStorePlugin = AWSDataStorePlugin(modelRegistration: AmplifyModels())
+            try Amplify.add(plugin: dataStorePlugin)
+            try Amplify.configure()
+            print("Amplify configured with DataStore plugin")
+        } catch {
+            print("Failed to initialize Amplify with \(error)")
+        }
+
+        let time: Temporal.DateTime = .now()
+        let post = Post(title: "title", content: "content", createdAt: time)
+        let id = post.id
+
+        Amplify.DataStore.save(post, completion: { result in
+            switch result {
+            case .success:
+                print("Post saved successfully!")
+                databaseCreationDone.fulfill()
+            case .failure(let error):
+                print("Error saving post \(error)")
+            }
+        })
+        wait(for: [databaseCreationDone], timeout: 10)
+
+        Amplify.reset()
+
+        let databaseRecreationDone = expectation(description: "Database remains")
+        do {
+            let dataStorePlugin = AWSDataStorePlugin(modelRegistration: AmplifyModels())
+            try Amplify.add(plugin: dataStorePlugin)
+            try Amplify.configure()
+            print("Amplify configured with DataStore plugin")
+        } catch {
+            print("Failed to initialize Amplify with \(error)")
+        }
+
+        Amplify.DataStore.query(Post.self, byId: id) { result in
+            switch result {
+            case .success(let postRes):
+                XCTAssertNotNil(postRes)
+                XCTAssertEqual(postRes!.id, id)
+                XCTAssertEqual(postRes!.title, "title")
+                XCTAssertEqual(postRes!.content, "content")
+                XCTAssertEqual(postRes!.createdAt, time)
```

Comment:
nit: yes the "postRes" is guarenteed to be not nil from line 71, but i'm still not convinced we should use force unwrapping here and should avoid it if we can. instead of 4 unwraps, you can do it once and provide a more meaningful message when it is `nil`
```
guard let queriedPost = postOptional else {
  XCTFail("could not retrieve post across Amplify re-configure")
  return
}
XCTAssertEqual(queriedPost.id, post.id)
```
you probably don't need to verify the post contents if we are already querying by Id

Reply:
true